File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int maxSize = 0;
        int windowSize = 0;
        int start = 0;
        HashSet<char> mySet = new HashSet<char>();
        for(int end = 0; end < s.Length; end++){
            char currentChar = s[end];
            
            while(mySet.Contains(currentChar)){
                mySet.Remove(s[start]);
                windowSize -= 1;
                start++;
            }
            
            mySet.Add(currentChar);
            windowSize += 1;
            maxSize = Math.Max(maxSize, windowSize);
        }
        return maxSize;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxSize = 0;
        int windowSize = 0;
        int start = 0;
        Set<Character> mySet = new HashSet<>();
        for(int end = 0; end < s.length(); end++){
            char currentChar = s.charAt(end);
            
            while(mySet.contains(currentChar)){
                mySet.remove(s.charAt(start));
                windowSize -= 1;
                start++;
            }
            
            mySet.add(currentChar);
            windowSize += 1;
            maxSize = Math.max(maxSize, windowSize);
        }
        return maxSize;
    }
}